src/common/stats/StatsManager.cpp

```diff
@@ -309,6 +321,214 @@ StatusOr<StatsManager::VT> StatsManager::readHisto(const std::string& counterNam
+    std::locale saved_locale = out.getloc();
+    out.imbue(std::locale::classic());
+    for (auto& index : sm.nameMap_) {
+        auto parsedName = sm.idParsedName_[index.second].get();
+        std::string name = parsedName != nullptr ? parsedName->name : index.first;
+        if (StatsManager::isStatIndex(index.second)) {
+            annotate(out, name, "gauge");
+
+            writeHead(out, name, {});
+            writeValue(out, StatsManager::readStats(index.second,
+                StatsManager::TimeRange::ONE_MINUTE, StatsManager::StatsMethod::AVG).value());
+            writeTail(out);
+        } else if (StatsManager::isHistoIndex(index.second)) {
+            annotate(out, name, "histogram");
+
+            writeHead(out, name, {}, "_count");
+            if (parsedName != nullptr) {
+                out << StatsManager::readStats(index.second,
+                    parsedName->range, StatsManager::StatsMethod::COUNT).value();
+            } else {
+                out << StatsManager::readStats(index.second,
+                    StatsManager::TimeRange::ONE_HOUR, StatsManager::StatsMethod::COUNT).value();
+            }
+            writeTail(out);
+
+            writeHead(out, name, {}, "_sum");
+            if (parsedName != nullptr) {
+                writeValue(out, StatsManager::readStats(index.second,
+                    parsedName->range, StatsManager::StatsMethod::SUM).value());
+            } else {
+                writeValue(out, StatsManager::readStats(index.second,
+                    StatsManager::TimeRange::ONE_HOUR, StatsManager::StatsMethod::SUM).value());
+            }
+            writeTail(out);
+
+            auto& p = sm.histograms_[StatsManager::physicalHistoIndex(index.second)];
+            std::lock_guard<std::mutex> lk(*p.first);
+            auto& hist = p.second;
+            auto last = hist->getMax();
+            auto cumulativeCount = 0;
+            auto diff = (hist->getMax() - hist->getMin()) /
+                static_cast<double>(hist->getNumBuckets());
+            double bound = hist->getMin() + diff;
+            for (std::size_t i = 0; i < hist->getNumBuckets(); ++i) {
+                writeHead(out, name, {}, "_bucket", "le", bound);
+                if (parsedName != nullptr) {
+                    cumulativeCount +=hist->getBucket(i)
+                        .count(static_cast<std::size_t>(parsedName->range));
+                } else {
+                    cumulativeCount += hist->getBucket(i)
+                        .count(static_cast<std::size_t>(StatsManager::TimeRange::ONE_HOUR));
+                }
+                out << cumulativeCount;
+                writeTail(out);
+                bound += diff;
+            }
+
+            if (!std::isinf(last)) {
+                writeHead(out, name, {}, "_bucket", "le", "+Inf");
+                out << cumulativeCount;
+                writeTail(out);
+            }
```

Comment:
I want to know what the metric name looks like after your formatting, could you give me some examples in comments?

Reply:
Such as below:
```
  1 # HELP put_kv_latency Record all histogram about nebula
     2 # TYPE put_kv_latency histogram
     3 put_kv_latency_count 0
     4 put_kv_latency_sum 0
     5 put_kv_latency_bucket{le="100.979904"} 0
     6 put_kv_latency_bucket{le="200.959808"} 0
     7 put_kv_latency_bucket{le="300.939712"} 0
     8 put_kv_latency_bucket{le="400.919616"} 0
     9 put_kv_latency_bucket{le="500.899520"} 0

......

110132 get_bound_latency_bucket{le="999900.020096"} 0
110133 get_bound_latency_bucket{le="1000000.000000"} 0
110134 get_bound_latency_bucket{le="+Inf"} 0
110135 # HELP add_edge_qps Record all gauge about nebula
110136 # TYPE add_edge_qps gauge
110137 add_edge_qps 0
110138 # HELP add_vertex_qps Record all gauge about nebula
110139 # TYPE add_vertex_qps gauge
110140 add_vertex_qps 0
110141 # HELP add_vertex_error_qps Record all gauge about nebula
110142 # TYPE add_vertex_error_qps gauge
110143 add_vertex_error_qps 0
```

Reply:
ok, how can I intuitively know which metric is from graph or storage?

Reply:
Need name with prefix such as `graphd_`.

Reply:
namespace, label or prefix, any way.  We can also make GraphStats/StorageStats to manage their labels or prefix.

Reply:
We can reference [other exporter](https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/exporters.md) formats. for nebula, like:
- `nebula_graphd_add_vertex_qps`
- `nebula_storaged_put_kv_latency`

Reply:
@laura-ding @critical27